packages/math/src/bn.test.ts

```diff
@@ -541,4 +541,35 @@ describe('Math - BN', () => {
   it('should return significant figures even if it exceeds the precision', () => {
     expect(bn('4000000').format({ precision: 1 })).toEqual('0.004');
   });
+
+  describe('precision validation', () => {
+    let consoleWarnSpy: jest.SpyInstance;
```

Comment:
We use `vitest`, you'll need to import `MockInstance` appropriately:

```suggestion
    let consoleWarnSpy: MockInstance;
```